assignment/assignment.hs

```diff
@@ -55,22 +74,29 @@ import Data.Char(intToDigit)
 -- 4 --
 -------
 -- Our application handles measurements in both millimeters (MM) and inches (IN)
--- write a data type named Measure that lets you specify a meansurement using
+-- write a data type named Measure that lets you specify a measurement using
 -- either unit. Be sure to derive Eq and Show.
 
---data Measure = ...
+data Measure = Millimeters Float | Inches Float
+  Deriving(Show, Eq)
+
 
 -- Now write a function that takes a measurement. If it is already in millimeters,
--- return it. Otherwise convert the measurement n inches to millimeters and
+-- return it. Otherwise convert the measurement in inches to millimeters and
 -- return a new metric Measure value. (Use 25.4 as the conversion factor)
 
 -- ensureMetric:: Measure -> Measure
 
+ensureMetric (Millimeters mm) = mm
+ensureMetric (Inches inc) = inc * 25.4
+
 -- Finally use `map` to take a list of Measure values, and return a new
 -- list where all measurements are in metric.
 
 -- listToMetric:: [ Measure ] -> [ Measure ]
 
+listToMetric list = map (ensureMetric) list
+
```

Comment:
The only small thing is that you'd be more idiomatic to write this as

~~~ hs
listToMetric = map ensureMetric
~~~

You can impress people at parties by referring to this process of removing a common parameter from both sides as "eta reduction".